CONTRIBUTING.md

```diff
@@ -1,20 +1,144 @@
-Thanks for wanting to contribute to Mozilla's Add-ons Linter! You rock! 😊
+Hi! Thanks for wanting to contribute to Mozilla's Add-ons Linter! You rock! 😊
 
-## Submitting a Pull Request
+The linter is used in various stages of [WebExtension](https://developer.mozilla.org/en-US/Add-ons/WebExtensions)
+development and publishing so by contributing you can make that process more awesome.
 
-If you're submitting a pull request, please reference the issue it closes
-inside your pull request. Keep your commits atomic; you should have one
-commit per issue solved.
+Here are links to all the sections in this document:
 
-Add tests for your code and make sure all existing tests pass. If the tests
-fail or you don't maintain 100% test coverage we won't be able to accept your
-pull request.
+<!-- If you change any of the headings in this document, remember to update the table of contents. -->
+<!-- To update the TOC, run the command `npm run gen-contributing-toc` from your root directory and you will auto generate a new TOC. -->
 
-### Tests
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
 
-Our tests include `eslint` style checks; these keep our code consistent.
-You can review the rules in the [`.eslintrc`][eslint] file.
 
-Please run the tests locally with `npm test` before you commit.
+- [Picking an issue](#picking-an-issue)
+- [Installation](#installation)
+  - [Develop all the things](#develop-all-the-things)
+  - [Run all application tests](#run-all-application-tests)
+    - [Run a single test](#run-a-single-test)
+    - [Debug a test](#debug-a-test)
+  - [Build addons-linter](#build-addons-linter)
+  - [Creating a pull request](#creating-a-pull-request)
+  - [Writing commit messages](#writing-commit-messages)
+  - [Tips](#tips)
 
-[eslint]: https://github.com/mozilla/addons-linter/blob/master/.eslintrc
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+
+# Picking an issue
+
+If you're looking for a small task to work on so you can get familiar with the
+process of contributing patches, have a read through these
+[good first bugs](https://github.com/mozilla/web-ext/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+bug%22).
+
+If you are already familiar with the process or want to contribute even more we appreciate any help on our [contrib: welcome](https://github.com/mozilla/addons-linter/issues?q=is%3Aissue+is%3Aopen+label%3A"contrib%3A+welcome) issues.
```

Comment:
Maybe something like this:

If you're already familiar with the project or would like take on something a little more challenging, please take a look at the [contrib: welcome](https://github.com/mozilla/addons-linter/issues?q=is%3Aissue+is%3Aopen+label%3A"contrib%3A+welcome) issues.